core/src/main/scala/kafka/log/Log.scala

```diff
@@ -345,13 +346,8 @@ class Log(@volatile var dir: File,
   }
 
   private def recoverLog() {
-    // if we have the clean shutdown marker, skip recovery
-    if(hasCleanShutdownFile) {
-      this.recoveryPoint = activeSegment.nextOffset
-      return
-    }
-
     // okay we need to actually recovery this log
```

Comment:
I think this comment is no longer true after this change. It's only if `unflushed.hasNext` returns true that we actually need to do recovery. Maybe we should only update the broker state in that case.

Also, it seems a bit weird that the state remains `RecoveringFromUncleanShutdown` even after recovery is complete for a given log. That seems correct according to the diagram in `BrokerState` though (I guess that simplifies things in case multiple logs need to be recovered).